bringing group mentoring process current (#5657)
kubernetes/community: mentoring/processes/group-mentoring.md

# Running a Contributor Ladder Growth Program 

> There are differences in ciricuulum for reviewer and Chair and/or Tech Lead 

Reviewer: Good catch

> There are differences in ciricuulum for reviewer and Chair and/or Tech Lead 

Reviewer 2: ```suggestion
There are differences in curriculum for reviewer and Chair and/or Tech Lead 
```

groups but the foundation of each are mostly the same unless otherwise noted 
below in the set up instructions. This difference is also reflected in our 
[community membership guidelines].

## Discovery of Needs

Cadence:  

> ContribEx sets a quarterly issue in k/community requesting SIGs to participate 

Reviewer 2: Suggest we do per-release rather than quarterly. Issue should probably precede the release by about 1mo.

Rolling:  
SIGs can approach ContribEx at anytime to start a group; file an issue in 
kubernetes/community or talk with us in #sig-contribex  

> ## Ciriculum Building

Reviewer 2: ```suggestion
## Curriculum Building
```

While not a fully structured program, some light planning is needed to make sure

> the ciriculum being taught is relevant for the group. Here are our base 

Reviewer 2: ```suggestion
the curriculum being taught is relevant for the group. Here are our base 
```

suggestions:
  

> *Reviewer*  

Reviewer 2: Add Overview of Release Cycle

> *Reviewer*  

Reviewer 2: Formatting suggestion: these should be bulleted lists.

Enhancements
Development
- Triage
- Reviews
- API Changes
- Testing
Code Freeze
Cherry-Picks/Branch Management

*Chair*

> KEPs++ 

Reviewer 2: ```suggestion
Enhancement Planning
```

Meetings and Organization

> Responsbilities and governance overview

Reviewer 2: ```suggestion
Responsibilities and governance overview
```

Consensus-building

> Communicating in our big world  

Reviewer 2: ```suggestion
Cross-project communication
```

*Chair/TL combo*
Chair + Reviewer

> [add more here]

Reviewer 2: Does this need to be a separate group? Are there people we would be considering for lead roles that were not already reviewers? I suspect that all content from the reviewer group would need to be covered (albeit briefly) in the Chair curriculum.

One of the best strategies is to have everyone work on something together, 
examples:
- reviewing guidelines for the SIG if there are none/updating guide
- CONTRIBUTING.md for the SIG if there isn't one/updating guide
- mid-size project
- a release cycle of features for the SIG, establishing better processes

## Setup and Outreach  

> 0. Need is established and SIG provides a mentor*

Reviewer 2: ```suggestion
0. Need is established and SIG provides a coordinator for the group and one or more additional mentors.
```

I'm not sure if "mentor" is the right name for this person, since we should have more than one mentor available to help the group.

1. ContribEx kicks off two issues, sets a target start and end date, and selects

> a coordinator

Reviewer 2: Would this be the person coordinating the group itself, or just a contribex contact?

Author: this would be the person coordinating the group itself, issue, outreach, slack, etc

  - issue #1: outreach issue to collect folks who are interested, issue will 
  close when there are the max participants needed
  - issue #2 setting up infra - template here: [TODO link to example]
2. ContribEx or SIG member posts the issue to the following places:
  - SIG and k-dev mailing lists
  - #diversity, #kubernetes-contributors, #sig-[name], and other slack channels
3. Coordinator arranges a doodle for the kick off meeting and days/times that 
would be best for the mentor and group for slack standups and touch points. Once conscensus is reached, coordinator schedules it.
4. Mentor creates template that serves as a reference guide for the group. See
a sig-node example here: https://hackmd.io/8i8prErgSOamUmEZDDZPWA
5. Program begins at the kick off meeting
  - [TODO add kick off template]

> 6. Feedback is continuous as you stand up weekly; however, Coorindator sets up

Reviewer 2: ```suggestion
6. Feedback is continuous as you stand up bi-weekly; however, coordinator sets up
```

Is this going to be a weekly or biweekly commitment? I found that attendance was quite poor for weekly, and bi-weekly was more doable.

Author: I think standing up weekly on slack as a check in is required with a bi-weekly call commitment/stand up  - wdyt?

a mid program check-in. In this check-in, members of the group will start their 

> application process for their roles. 

Reviewer 2: ```suggestion
application process for their roles and any group members who have not had the time
to participate will be given the opportunity to drop out.
```

  - [TODO add community membership template]
7. By this point, folks should know where they stand. At the end of the program,
 mentors and/or individuals will start the PR process in OWNERS files for those 

> who are ready  

Reviewer 2: For Chair/TL roles, there aren't really OWNERS files to update in order to kick off a transition. This might be pointing to a gap in SIG leadership succession processes.

Author: there is sigs.yaml

> [community membership guidelines]: 

Reviewer 2: ```suggestion
[community membership guidelines]: community-membership.md
```
